cub/test/catch2_test_device_run_length_encode_non_trivial_runs.cu

```diff
@@ -368,3 +388,98 @@ C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns does not run out of memory", "[d
+  // Number of non-trivial runs is number of full(!) three-item (i.e., even, odd, odd) groups
+  const auto num_uniques = num_items / 3;
+
+  // Prepare helper to check the unique items being written: we expect the i-th item corresponding to value i
+  auto check_offset_out_helper = detail::large_problem_test_helper(num_uniques);
+  auto expected_offsets_it     = cuda::make_transform_iterator(counting_it, run_index_to_offset_op{});
+  auto check_offset_out_it     = check_offset_out_helper.get_flagging_output_iterator(expected_offsets_it);
+
+  // Prepare helper to check the run-lengths being written: i-th item corresponding to value i
+  // We repeat each number once for the first num_small_runs number of items and all subsequent numbers twice
+  auto check_run_length_out_helper = detail::large_problem_test_helper(num_uniques);
+  auto expected_run_lengths_it     = cuda::make_constant_iterator(run_length_type{2});
+  auto check_run_length_out_it     = check_run_length_out_helper.get_flagging_output_iterator(expected_run_lengths_it);
+
+  // Allocate memory for the number of expected unique items
+  c2h::device_vector<offset_type> out_num_runs(1);
+
+  // Run algorithm under test
+  run_length_encode(input_item_it, check_offset_out_it, check_run_length_out_it, out_num_runs.begin(), num_items);
+
+  // Verify result
+  REQUIRE(out_num_runs[0] == num_uniques);
+  check_offset_out_helper.check_all_results_correct();
+  check_run_length_out_helper.check_all_results_correct();
+}
+catch (std::bad_alloc& e)
+{
+  std::cerr << "Caught bad_alloc: " << e.what() << std::endl;
+}
+
+C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns works for large runs of equal items",
+         "[device][run_length_encode][skip-cs-initcheck][skip-cs-racecheck][skip-cs-synccheck][!mayfail]",
+         offset_types)
+try
+{
+  using offset_type     = typename c2h::get<0, TestType>;
+  using run_length_type = offset_type;
+  using item_t          = char;
+
+  CAPTURE(c2h::type_name<offset_type>(), c2h::type_name<run_length_type>(), c2h::type_name<item_t>());
+
+  const auto num_items = detail::make_large_offset<offset_type>();
+
+  constexpr auto num_uniques               = offset_type{2};
+  constexpr run_length_type first_run_size = 200;
+  const run_length_type second_run_size    = static_cast<run_length_type>(num_items) - first_run_size;
+
+  // First run is a small run of equal items
+  auto small_segment_it = cuda::make_constant_iterator(item_t{3});
+  // Second run is a very large run of equal items
+  auto large_segment_it = cuda::make_constant_iterator(item_t{42});
+  auto input_item_it    = detail::make_concat_iterators_op(small_segment_it, large_segment_it, first_run_size);
+
+  // Allocate some memory for the results
+  c2h::device_vector<offset_type> offsets_out(num_uniques, offset_type{42});
+  c2h::device_vector<run_length_type> run_lengths_out(num_uniques, run_length_type{42});
```

Comment:
Suggestion: Both of these are outputs and must be written, so you may also leave them uninitialized:
```suggestion
  c2h::device_vector<offset_type> offsets_out(num_uniques, thrust::no_init);
  c2h::device_vector<run_length_type> run_lengths_out(num_uniques, thrust::no_init);
```